### PR TITLE
Correct excessive indentation in StaticHeader::setup

### LINKED ISSUE #1575: Correct excessive indentation in StaticHeader::setup
Each match arm for ```BDA::read(f)``` has a corresponding expression which is nested one level deeply as

```
{
   match ...
}
```

Those extra braces should be eliminated. This extra nesting is apparently done in order to give the comments a place, but the comments shouldn't force the code that way.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #1575.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
I'll go for a more global cleanup then.

**Comment by mulkieran** [comment]
Thanks! I had a thought. What happens if you take out some of this unnecessary or wrong comments and just run rustfmt? Will rustfmt eliminate the extra braces on its own?

**Comment by GuillaumeGomez** [comment]
Well, I find those comments useful. And I was able to improve the situation a bit. :)

File: src/engine/strat_engine/backstore/metadata/bda.rs [hunk 1]
```diff
@@ -327,111 +327,103 @@ impl StaticHeader {
...
+                        BDA::write(f, &buf_loc_2, MetadataLocation::First)?;
+                        Ok(Some(loc_2))
                     }
-                    Err(e) => {
-                        // Unable to determine if location 2 has a signature, but location 1 did,
-                        // but it got an error, lets return the error instead as this appears to
-                        // be a stratis device that has gotten in a bad state.
-                        Err(e)
+                },
+                // Re-write copy 2
```
Comment (mulkieran) [inline src/engine/strat_engine/backstore/metadata/bda.rs]: Do please drop this comment, it overstates what the code actually does.

File: src/engine/strat_engine/backstore/metadata/bda.rs [hunk 2]
```diff
@@ -327,111 +327,103 @@ impl StaticHeader {
...
                 }
-            }
-            (Err(_), Ok(buf_loc_2)) => {
-                // Copy 2 read OK, 1 resulted in IO Error
-                match StaticHeader::sigblock_from_buf(&buf_loc_2) {
-                    Ok(loc_2) => {
-                        if loc_2.is_some() {
-                            BDA::write(f, &buf_loc_2, MetadataLocation::First)?;
-                        }
+                // Re-write copy 1
```
Comment (mulkieran) [inline src/engine/strat_engine/backstore/metadata/bda.rs]: Please drop this comment as well.

File: src/engine/strat_engine/backstore/metadata/bda.rs [hunk 3]
```diff
@@ -327,111 +327,103 @@ impl StaticHeader {
...
+                        Ok(Some(loc_2))
                     }
-                    Err(e) => {
-                        // Unable to determine if location 2 has a signature, but location 1 did,
-                        // but it got an error, lets return the error instead as this appears to
-                        // be a stratis device that has gotten in a bad state.
-                        Err(e)
+                },
+                // Re-write copy 2
+                (Ok(loc_1), Err(loc_2)) => {
```
Comment (mulkieran) [inline src/engine/strat_engine/backstore/metadata/bda.rs]: You can drop extra curly braces around the if-expression here.

File: src/engine/strat_engine/backstore/metadata/bda.rs [hunk 4]
```diff
@@ -327,111 +327,103 @@ impl StaticHeader {
...
-            }
-            (Err(_), Ok(buf_loc_2)) => {
-                // Copy 2 read OK, 1 resulted in IO Error
-                match StaticHeader::sigblock_from_buf(&buf_loc_2) {
-                    Ok(loc_2) => {
-                        if loc_2.is_some() {
-                            BDA::write(f, &buf_loc_2, MetadataLocation::First)?;
-                        }
+                // Re-write copy 1
+                (Err(loc_1), Ok(loc_2)) => {
```
Comment (mulkieran) [inline src/engine/strat_engine/backstore/metadata/bda.rs]: You can drop extra curly braces around the if-expression here.

File: src/engine/strat_engine/backstore/metadata/bda.rs (hunk 3 above)
Comment (GuillaumeGomez) [reply src/engine/strat_engine/backstore/metadata/bda.rs]: I tried, `cargo fmt` doesn't want to. 😆

File: src/engine/strat_engine/backstore/metadata/bda.rs (hunk 4 above)
Comment (GuillaumeGomez) [reply src/engine/strat_engine/backstore/metadata/bda.rs]: Same.

**Comment by GuillaumeGomez** [comment]
Updated!

**Comment by mulkieran** [comment]
ok to test

**Pull request merged**